File: matiasmasso/Mat.NET/DTO/DTOContactTel.cs
using MatHelperStd;
using System;

namespace DTO
{
    public class DTOContactTel : DTOBaseTel
    {
        public Cods Cod { get; set; }
        public int Ord { get; set; }
        public DTOCountry Country { get; set; }
        public bool Privat { get; set; }

        public enum IncludePrefix
        {
            None,
            IfNotCurrent,
            Always
        }


        public enum Cods
        {
            NotSet,
            tel,
            fax,
            movil,
            email
        }

        public DTOContactTel() : base()
        {
            base.ObjCod = DTOBaseTel.ObjCods.Tel;
        }

        public DTOContactTel(Guid oGuid) : base(oGuid)
        {
            base.ObjCod = DTOBaseTel.ObjCods.Tel;
        }

        public static DTOContactTel Factory(DTOContact oContact, DTOContactTel.Cods oCod = DTOContactTel.Cods.tel, string value = "")
        {
            DTOContactTel retval = new DTOContactTel();
            {
                var withBlock = retval;
                withBlock.Cod = oCod;
                withBlock.Country = DTOAddress.Country(oContact.Address);
                withBlock.Value = TextHelper.LeaveJustNumericDigits(value);
            }
            return retval;
        }

        public static string CleanValue(DTOContactTel oContactTel)
        {
            string retval = "";
            if (oContactTel != null)
                retval = TextHelper.LeaveJustNumericDigits(oContactTel.Value);
            return retval;
        }

        public static string Formatted(DTOContactTel value)
        {
            string retval = "";
            if (value != null)
            {
                retval = value.Value;
                if (retval.Length > 6)
                {
                    retval = retval.Insert(6, ".");
                    retval = retval.Insert(3, ".");
                }
                if (value.Country != null)
                {
                    if (!DTOArea.isEsp(value.Country))
                    {
                        if (value.Country.PrefixeTelefonic.isNotEmpty())
                            retval = "(+" + value.Country.PrefixeTelefonic + ") " + retval;
                    }
                }
            }
            return retval;
        }

        public static string HtmlLink(DTOContactTel value)
        {
            string retval = value.Value;
            if (value.Country != null)
            {
                if (value.Country.PrefixeTelefonic.isNotEmpty())
                    retval = string.Format("+{0}{1}", value.Country.PrefixeTelefonic, value.Value);
            }
            return retval;
        }

        public static string Html5Formatted(DTOContactTel oContactTel)
        {
            string retval = "";
            if (oContactTel != null)
            {
                if (oContactTel.Value.isNotEmpty())
                {
                    string sTelnum = System.Text.RegularExpressions.Regex.Replace(oContactTel.Value, "[^0-9]", "");
                    if (oContactTel.Country != null && oContactTel.Country.PrefixeTelefonic.isNotEmpty())
                        retval = string.Format("tel:+{0}{1}", oContactTel.Country.PrefixeTelefonic, sTelnum);
                    else
                        retval = string.Format("tel:{0}", sTelnum);
                }
            }
            return retval;
        }
    }
}

File: matiasmasso/Mat.NET/DT2/DTOContactTel.vb
Public Class DTOContactTel
    Inherits DTOBaseTel

    Property cod As Cods
    Property ord As Integer
    Property country As DTOCountry
    Property privat As Boolean

    Public Enum IncludePrefix
        None
        IfNotCurrent
        Always
    End Enum


    Public Enum Cods
        NotSet
        tel
        fax
        movil
        email
    End Enum

    Public Sub New()
        MyBase.New()
        MyBase.objCod = DTOBaseTel.ObjCods.Tel
    End Sub

    Public Sub New(oGuid As Guid)
        MyBase.New(oGuid)
        MyBase.objCod = DTOBaseTel.ObjCods.Tel
    End Sub

    Shared Function Factory(oContact As DTOContact, Optional oCod As DTOContactTel.Cods = DTOContactTel.Cods.tel, Optional value As String = "") As DTOContactTel
        Dim retval As New DTOContactTel
        With retval
            .cod = oCod
            .country = DTOAddress.Country(oContact.address)
            .value = TextHelper.LeaveJustNumbericDigits(value)
        End With
        Return retval
    End Function

    Shared Function CleanValue(oContactTel As DTOContactTel) As String
        Dim retval As String = ""
        If oContactTel IsNot Nothing Then
            retval = TextHelper.LeaveJustNumbericDigits(oContactTel.value)
        End If
        Return retval
    End Function

    Shared Shadows Function Formatted(value As DTOContactTel) As String
        Dim retval As String = ""
        If value IsNot Nothing Then
            retval = value.value
            If retval.Length > 6 Then
                retval = retval.Insert(6, ".")
                retval = retval.Insert(3, ".")
            End If
            If value.country IsNot Nothing Then
                If Not DTOArea.IsEsp(value.country) Then
                    If value.country.prefixeTelefonic > "" Then
                        retval = "(+" & value.country.prefixeTelefonic & ") " & retval
                    End If
                End If
            End If
        End If
        Return retval
    End Function

    Shared Function HtmlLink(value As DTOContactTel) As String
        Dim retval As String = value.value
        If value.country IsNot Nothing Then
            If value.country.prefixeTelefonic > "" Then
                retval = String.Format("+{0}{1}", value.country.prefixeTelefonic, value.value)
            End If
        End If
        Return retval
    End Function

    Shared Function Html5Formatted(oContactTel As DTOContactTel) As String
        Dim retval As String = ""
        If oContactTel IsNot Nothing Then
            If oContactTel.value > "" Then
                Dim sTelnum As String = System.Text.RegularExpressions.Regex.Replace(oContactTel.value, "[^0-9]", "")
                If oContactTel.country IsNot Nothing AndAlso oContactTel.country.prefixeTelefonic > "" Then
                    retval = String.Format("tel:+{0}{1}", oContactTel.country.prefixeTelefonic, sTelnum)
                Else
                    retval = String.Format("tel:{0}", sTelnum)
                End If
            End If
        End If
        Return retval
    End Function
End Class
